src/util/signature_utils.js

```diff
@@ -203,7 +203,7 @@ function _aws_request(req, region, service) {
     const v2_signature = _.isUndefined(region) && _.isUndefined(service);
     const u = url.parse(req.originalUrl.replace(/%2F/g, '/'), true);
     const pathname = service === 's3' ?
-        u.pathname :
+        u.pathname.split('/').map(c => AWS.util.uriEscape(decodeURIComponent(c))).join('/') :
```

Comment:
can you please just add a comment that explains this logix and why its needed? it's going to be hard to remember the details when reading the code later. tnx